contracts/Web3Entry.sol

```diff
@@ -189,15 +199,42 @@ contract Web3Entry is Web3EntryBase {
         uint256 permissionId
     ) internal view override {
         address owner = ownerOf(characterId);
-        require(
-            msg.sender == owner ||
-                (tx.origin == owner && msg.sender == periphery) ||
-                (((_operatorsPermission4NoteBitMap[characterId][noteId][msg.sender] >>
-                    permissionId) & 1) ==
-                    1 ||
-                    _operatorsPermission4NoteBitMap[characterId][noteId][msg.sender] == 0),
-            "NotEnoughPermissionForThisNote"
-        );
+        if (msg.sender == owner) {
+            // check if it's owner
+        } else if (tx.origin == owner && msg.sender == periphery) {
+            // check if it's periphery
+        } else if (_operatorsPermission4NoteBitMap[characterId][noteId][msg.sender] == 0) {
+            if (_checkBit(_operatorsPermissionBitMap[characterId][msg.sender], permissionId)) {
+                // check if it has operator permission for this method and if it's open to all notes
+            } else {
+                revert("NotEnoughPermissionForThisNote");
+            }
+        } else if (
+            ((_operatorsPermission4NoteBitMap[characterId][noteId][msg.sender] >> permissionId) &
+                1) == 1
+        ) {
+            // check if it has note permission
+        } else {
+            // if it doesn't have corresponding permission,
+            revert("NotEnoughPermissionForThisNote"); // then this caller is nothing, we need to revert.
+        }
+    }
+
+    /**
+     * @dev _bitmapFilter unsets bits of non-existent permission IDs to zero. These unset permission IDs are 
+     meaningless now, but they are reserved for future use, so it's best to leave them blank and avoid messing
+      up with future methods.
+     */
+    function _bitmapFilter(uint256 bitmap) internal returns (uint256) {
+        uint256 filteredBitmap = bitmap & OP.ALLOWED_PERMISSION_BITMAP_MASK;
+        return filteredBitmap;
```

Comment:
better  to remove the definition of filteredBitmap ? just use `return bitmap & OP.ALLOWED_PERMISSION_BITMAP_MASK;`